frontend/src/metabase-lib/mocks.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  Card,
+  SavedCard,
+  UnsavedCard,
+  StructuredDatasetQuery,
+  NativeDatasetQuery,
+} from "metabase-types/types/Card";
+
+import {
+  SAMPLE_DATABASE,
+  ORDERS,
+  metadata,
+} from "__support__/sample_database_fixture";
+
+import Question from "./lib/Question";
+import NativeQuery from "./lib/queries/NativeQuery";
+import StructuredQuery from "./lib/queries/StructuredQuery";
+
+type NativeSavedCard = SavedCard<NativeDatasetQuery>;
+type NativeUnsavedCard = UnsavedCard<NativeDatasetQuery>;
+type StructuredSavedCard = SavedCard<StructuredDatasetQuery>;
+type StructuredUnsavedCard = UnsavedCard<StructuredDatasetQuery>;
+
+const BASE_GUI_QUESTION: StructuredUnsavedCard = {
+  display: "table",
+  visualization_settings: {},
+  dataset_query: {
+    type: "query",
+    database: SAMPLE_DATABASE?.id,
+    query: {
+      "source-table": ORDERS.id,
+    },
+  },
+};
+
+const BASE_NATIVE_QUESTION: NativeUnsavedCard = {
+  display: "table",
+  visualization_settings: {},
+  dataset_query: {
+    type: "native",
+    database: SAMPLE_DATABASE?.id,
+    native: {
+      query: "select * from orders",
+      "template-tags": {},
+    },
+  },
+};
+
+const SAVED_QUESTION = {
+  id: 1,
+  name: "Q1",
+  description: "",
+  collection_id: null,
+};
+
+export function getQuestion(card: Partial<Card>) {
+  return new Question(
+    {
+      ...BASE_GUI_QUESTION,
+      display: "table",
+      visualization_settings: {},
+      ...card,
+    },
+    metadata,
+  );
+}
+
+export function getAdHocQuestion(card?: Partial<StructuredUnsavedCard>) {
+  return getQuestion({ ...BASE_GUI_QUESTION, ...card });
+}
+
+export function getCleanStructuredQuestion(
+  card?: Partial<StructuredUnsavedCard>,
+) {
+  const question = getAdHocQuestion(card);
+
+  const query = (question.query() as StructuredQuery).setSourceTableId(
```

Comment:
wdyt
```
const query = question.query();
if (query instanceof StructuredQuery) {
  query.setSourceTableId(undefined);
}
```

Reply:
And we would need to fix `setSourceTableId` to accept `undefined`

Reply:
Yeah, sounds good, thanks

Reply:
👍